punchcut/src/error.rs

```diff
@@ -0,0 +1,52 @@
+use read_fonts::{types::GlyphId, ReadError};
+
+use std::fmt;
+
+/// Errors that may occur when loading glyphs.
+#[derive(Clone, Debug)]
+pub enum Error {
+    /// No viable sources were available.
+    NoSources,
+    /// The requested glyph was not present in the font.
+    GlyphNotFound(GlyphId),
+    /// Exceeded a recursion limit when loading a glyph.
+    RecursionLimitExceeded(GlyphId),
+    /// Error occured during hinting.
+    HintingFailed(GlyphId),
+    /// An anchor point had invalid indices.
+    InvalidAnchorPoint(GlyphId),
+    /// Error occured when reading font data.
+    Read(ReadError),
+}
+
+impl From<ReadError> for Error {
+    fn from(e: ReadError) -> Self {
+        Self::Read(e)
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::NoSources => write!(f, "No glyph sources are available for the given font"),
+            Self::GlyphNotFound(gid) => write!(f, "Glyph {} was not found in the given font", gid),
+            Self::RecursionLimitExceeded(gid) => write!(
+                f,
+                "Recursion limit exceeded when loading composite component {}",
+                gid
+            ),
+            Self::HintingFailed(gid) => write!(f, "Bad hinting bytecode for glyph {}", gid),
```

Comment:
Feature gate this one.

Also, we should provide any additional information we have. Perhaps we know _something_ about what went wrong?

Reply:
It doesn't currently expose any details about what went wrong. That's been a todo of mine for a while.